azure-iot-hub-devicesdk/samples/authentication_providers.py

```diff
@@ -0,0 +1,37 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import os
+from azure.iot.hub.devicesdk.device_client import DeviceClient, ModuleClient
+
+from azure.iot.hub.devicesdk.auth.authentication_provider_factory import (
+    from_shared_access_signature,
+    from_connection_string,
+    from_environment,
+)
+
+# To understand Authentication Providers, it is beneficial to be familiar with the Azure IoT Hub Security model:
+# https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security
+
+# Authentication providers are used by the client to get the keys or certificates used
+# to connect securely to an Azure IoT hub.
+
+# An Authentication Provider can be created from a few different things:
+# - A shared access signature:
+sas_auth_provider = from_shared_access_signature(os.getenv("IOTHUB_DEVICE_SAS_STRING"))
+
+# - A symmetric key:
+key_auth_provider = from_connection_string("IOTHUB_DEVICE_CONNECTION_STRING")
+
+# - Pre-defined environment variables (this is especially useful when running on Azure IoT Edge)
```

Comment:
>especially useful [](start = 47, length = 17)

I think you copy/pasted this phrase from me, but I would say "this only works when running as an Azure IoT Edge Module"